auth-web/src/components/pay/eft/ShortNameRefund.vue

```diff
@@ -25,46 +28,266 @@
       results: [],
-      totalResults: 0,
+      totalResults: 1,
       filters: {
         pageNumber: 1,
         pageLimit: 5
       },
       loading: false,
       options: _.cloneDeep(DEFAULT_DATA_OPTIONS),
-      expanded: []
+      expanded: [],
+      currentEftRefund: null,
+      isEftRefundApprover: computed(() => userStore.currentUser.roles.includes(Role.EftRefundApprover))
     })
+    const confirmationDialog: Ref<InstanceType<typeof ModalDialog>> = ref(null)
+    const headers = [
+      {
+        col: 'createdName',
+        hasFilter: false,
+        width: '300px',
+        value: 'Initiated by'
+      },
+      {
+        col: 'casSupplierNumber',
+        hasFilter: false,
+        width: '210px',
+        value: 'Supplier Record Number'
+      },
+      {
+        col: 'refundAmount',
+        hasFilter: false,
+        width: '260px',
+        value: 'Refund Amount'
+      },
+      {
+        col: 'actions',
+        hasFilter: false,
+        value: 'Actions',
+        width: '200px'
+      }
+    ]
 
-    const isLinked = computed<boolean>(() => {
-      return state.totalResults > 0 || state.loading
-    })
+    async function approveRefund (item) {
+      const shortNameRefund = {
+        status: EFTRefundType.APPROVED
+      }
+      try {
+        await PaymentService.patchEFTRefund(item.id, shortNameRefund)
+        await loadTransactions(props.shortNameDetails.id)
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Failed to approve refund.', error)
+      }
+    }
+
+    function declineRefund (item) {
+      state.currentEftRefund = item
+      confirmationDialog.value.open()
+    }
+
+    async function dialogDecline () {
+      const shortNameRefund = {
+        status: EFTRefundType.DECLINED,
+        decline_reason: state.declineReason
+      }
+      try {
+        await PaymentService.patchEFTRefund(state.currentEftRefund.id, shortNameRefund)
+        await loadTransactions(props.shortNameDetails.id)
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Failed to decline refund.', error)
+      }
+      confirmationDialog.value.close()
+      state.currentEftRefund = null
+    }
+
+    watch(() => props.shortNameDetails, () => {
+      return loadTransactions(props.shortNameDetails.id)
+    }, { immediate: true, deep: true })
+
+    async function loadTransactions (shortnameId: string): Promise<void> {
+      try {
+        state.loading = true
+        const response = await PaymentService.getEFTShortnameHistory(shortnameId, state.filters)
+        if (response?.data) {
+          const filteredResults = response.data.items.filter(
+            (item) => item.transactionType === ShortNameHistoryType.SN_REFUND_PENDING_APPROVAL
+          )
+
+          for (const item of filteredResults) {
+            const eftRefundId = item.eftRefundId
+            const eftRefund = await PaymentService.getEFTRefund(eftRefundId)
```

Comment:
synchronous is ok here? guessing there aren't alot of results? otherwise it's 1 by 1

Reply:
@seeker25 Haha you already did this, genius

![Screenshot 2024-10-16 at 2 26 52 PM](https://github.com/user-attachments/assets/b3ec24b3-4324-4d7f-a2e5-5bee9ab9a488)

Reply:
We may need to add a filter for short_name_id since the path isn't by short name identifier